bin/dropletBarcodePlot.R

```diff
@@ -66,38 +137,32 @@ barcode_rank_plot <- function(br.out, roryk_total_cutoff, knee, inflection, name
     ggtitle(name) + ylab('Count') + xlab('Rank') + theme(legend.position = "none")
 }
 
-# Plot the different plots and threshold statistics alongside one another
+# Sort barcodes by descending frequency
 
-raw_barcodes <- read.delim(barcode_file, header = FALSE)
-result_matrix <- read10xCounts(result_matrix)  
-processed_barcode_counts <- data.frame(V1 = colData(result_matrix)$Barcode, V2=colSums(assays(result_matrix)[[1]]))
-processed_barcode_counts <- processed_barcode_counts[order(processed_barcode_counts$V2, decreasing = TRUE), ]
-
-barcode_results <- list(
-    raw = raw_barcodes,
-    processed = processed_barcode_counts
-)
+barcode_counts <- barcode_counts[order(barcode_counts$V2, decreasing = TRUE), ]
 
-plots <- lapply(names(barcode_results), function(name){
+roryk_count_cutoff <- pick_roryk_cutoff(barcode_counts$V2, opt$roryk_multiplier)
   
-  barcodes <- barcode_results[[name]]  
+# Run dropletUtils' barcodeRanks to get knee etc
+br.out <- barcodeRanks(t(barcode_counts[,2,drop=FALSE]))
   
-  # Get the roryk cutoff
-  roryk_count_cutoff <- pick_roryk_cutoff(barcodes$V2)
-  
-  # Run dropletUtils' barcodeRanks to get knee etc
-  br.out <- barcodeRanks(t(barcodes[,2,drop=FALSE]))
-  
-  dropletutils_knee <- metadata(br.out)$knee
-  dropletutils_inflection <- metadata(br.out)$inflection
+dropletutils_knee <- metadata(br.out)$knee
+dropletutils_inflection <- metadata(br.out)$inflection
+
+plot_label <- paste(format(nrow(barcode_counts), big.mark = ','), 'cell barcodes')
+if ((! is.na(opt$label)) && opt$label != ''){
+  plot_label <- paste0(opt$label, ': ', plot_label)
+}
   
-  list(
-    dropletutils = barcode_rank_plot(br.out, roryk_count_cutoff, dropletutils_knee, dropletutils_inflection, name = paste(label, name)),
-    roryk = barcode_density_plot(barcodes$V2, roryk_count_cutoff, dropletutils_knee, dropletutils_inflection, name = paste(label, name))
-  )
-})
-names(plots) <- names(barcode_results)
+plots <- list(
+  dropletutils = barcode_rank_plot(br.out, roryk_count_cutoff, dropletutils_knee, dropletutils_inflection, name = plot_label),
+  roryk = barcode_density_plot(barcode_counts$V2, roryk_count_cutoff, dropletutils_knee, dropletutils_inflection, name = '   ')
+)
 
-png(width = 1000, height = 800, file=output_plot_file)
-grid.arrange(plots$raw$dropletutils, plots$raw$roryk, plots$processed$dropletutils, plots$processed$roryk, nrow=2)
+# Create output plot
+png(width = 1000, height = 600, file=opt$output_plot)
+grid.arrange(plots$dropletutils, plots$roryk, nrow=1)
 dev.off()
+
+# Return calculated thresholds
+write.table(data.frame(dropletutils_knee = dropletutils_knee, dropletutils_inflection = dropletutils_inflection, roryk=roryk_count_cutoff), file = opt$output_thresholds, row.names = FALSE, quote = FALSE)
```

Comment:
do you need this table in production or only for the test env?

Reply:
We probably don't nee that table at all.

Reply:
We probably don't nee that table at all.